store/layer_generators/main.go

```diff
@@ -92,10 +115,20 @@ func ExtractStoreMetadata() StoreMetadata {
 					if _, ok := topLevelFunctions[methodName]; ok {
 						params := []Param{}
 						results := []string{}
-
+						paramsToTrace := map[string]bool{}
 						ast.Inspect(method.Type, func(expr ast.Node) bool {
```

Comment:
This ast.Inspect handler for x.Name.Name == "store" and the else if condition below looks suspiciously similar. Any chance we could refactor these to a separate function ?